bundles/net.tourbook/src/net/tourbook/map2/view/DialogMap2ExportViewImage.java

```diff
@@ -0,0 +1,559 @@
+/*******************************************************************************
+ * Copyright (C) 2021 Frédéric Bard
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.map2.view;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.common.UI;
+import net.tourbook.common.util.Util;
+import net.tourbook.map2.Messages;
+import net.tourbook.ui.FileCollisionBehavior;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.layout.PixelConverter;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageLoader;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.FileDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+
+public class DialogMap2ExportViewImage extends TitleAreaDialog {
```

Comment:
Yes, png have different values but they don't affect the quality of the image ([source](https://javadoc.scijava.org/Eclipse/org/eclipse/swt/graphics/ImageLoader.html#compression))

Reply:
Fixed. I added a scale to choose the compression value
![image](https://user-images.githubusercontent.com/25235665/105533636-a573f300-5ca9-11eb-826e-5136f19b8216.png)

Reply:
`50 %` not `50 ?`

Reply:
After testing the export dialog, I think that it would be helpfull to know, what this number means. It is not the compression in % it is the image quality as described in the javadoc

`
When saving jpeg files, the value is from 1 to 100,where 1 is very high compression but low quality, and 100 isno compression and high quality; default is 75. 
`

Reply:
> After testing the export dialog, I think that it would be helpfull to know, what this number means. It is not the compression in % it is the image quality as described in the javadoc
> `When saving jpeg files, the value is from 1 to 100,where 1 is very high compression but low quality, and 100 isno compression and high quality; default is 75. `

I also put a tooltip that repeats the java doc, did it not work for you ?
![image](https://user-images.githubusercontent.com/25235665/105609489-f0077500-5d66-11eb-97d0-e06f7fb621ac.png)
Also, the java doc says it's the compression value. It affects the quality but the parameter is called "compression"
Let me know what you prefer

![image](https://user-images.githubusercontent.com/25235665/105609531-2e049900-5d67-11eb-9def-5ee4426a407d.png)

Reply:
> > After testing the export dialog, I think that it would be helpfull to know, what this number means. It is not the compression in % it is the image quality as described in the javadoc
> > `When saving jpeg files, the value is from 1 to 100,where 1 is very high compression but low quality, and 100 isno compression and high quality; default is 75. `
> 
> I also put a tooltip that repeats the java doc, did it not work for you ?

I havn't seen the tooltip. I already discovered by other users, that tooltips are not read because they are at the wrong location. To force that a tooltip is read (at least visible), attach the tooltip to the control and not to the label or to both, in this case to 3 controls (label, number, scale)

![mt-jpp-compression-tooltip](https://user-images.githubusercontent.com/1283445/105624105-6ebfe900-5e1f-11eb-9f1b-0c11d167d6fc.png)

Win10 has no autowrap for tooltips so you should insert hard coded line breaks that a tooltip is easier to read.


I would still suggest, to show the image quality and not the compression rate. The FastStone image viewer is also showing the image quality for jpeg's and allows the user to enter a value in the spinner control not only a scale control.

One realy bad thing in FastStone image viewer is, it do not support mnemonic's in dialogs however keyboad shortcuts in other locations

![mt-jpp-compression](https://user-images.githubusercontent.com/1283445/105624182-f60d5c80-5e1f-11eb-98f2-d5b18b465e8a.png)

Can you also show jpeg... in the listbox with uppercase letters?

Reply:
In labels, we should use JPEG instead of JPG, see https://www.keycdn.com/support/difference-between-jpg-and-jpeg but for the file extension we should use the common jpg

Reply:
In the listbox we should display `JPEG, JPG`, in other places only `JPEG`

Reply:
> In the listbox we should display JPEG, JPG, in other places only JPEG

>I would still suggest, to show the image quality and not the compression rate.

>To force that a tooltip is read (at least visible), attach the tooltip to the control and not to the label or to both, in this case to 3 controls (label, number, scale)

Fixed

![image](https://user-images.githubusercontent.com/25235665/105659048-0f94bf80-5e85-11eb-8a60-de5b62b7d43a.png)

Reply:
I just found that exporting the map as a JPEG image with a quality of 12 or 75 or 100 etc.... doesn't affect the image file in Linux but it works in Windows.
Do you see the same issue when you test on your ubuntu ?

Reply:
> 
> 
> > In the listbox we should display JPEG, JPG, in other places only JPEG
> 
> > I would still suggest, to show the image quality and not the compression rate.
> 
> > To force that a tooltip is read (at least visible), attach the tooltip to the control and not to the label or to both, in this case to 3 controls (label, number, scale)
> 
> Fixed
> 
> ![image](https://user-images.githubusercontent.com/25235665/105659048-0f94bf80-5e85-11eb-8a60-de5b62b7d43a.png)

Almost perfect, JPEG is always written in uppercase https://en.wikipedia.org/wiki/JPEG

Reply:
> 
> 
> I just found that exporting the map as a JPEG image with a quality of 12 or 75 or 100 etc.... doesn't affect the image file in Linux but it works in Windows.
> Do you see the same issue when you test on your ubuntu ?

Yes, Ubuntu do not show any differences, all the image files have the same size

Reply:
When `Quality` cannot be set on Linux, it should be disabled that nobody can complain that it is not working

Reply:
Disabled when on Linux. Available in my PR.

Reply:
Also, I have created an Eclipse bug [here](https://bugs.eclipse.org/bugs/show_bug.cgi?id=570646) with a code snippet to reproduce

Reply:
> 
> 
> Also, I have created an Eclipse bug [here](https://bugs.eclipse.org/bugs/show_bug.cgi?id=570646) with a code snippet to reproduce

I think you should attach the snippet to the bug otherwise the bug has a dependency on your gitlab resouce and you never know when this will be fixed, in a week, month, year or 10 years?

Reply:
True, maybe I should have put a snippet. My previous bugs were resolved fast so I assume this one could be resolved fast too but maybe not
https://bugs.eclipse.org/bugs/show_bug.cgi?id=553240
https://bugs.eclipse.org/bugs/show_bug.cgi?id=547489

Reply:
The bug could be set to won't fix or you should fix it by your own https://bugs.eclipse.org/bugs/show_bug.cgi?id=539211